File: Leet Code/53. Maximum Subarray/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSub = nums[0]
        current_sum = 0
        for n in nums : 
            if current_sum < 0:
                current_sum = 0
            current_sum = current_sum + n
            maxSub = max(maxSub, current_sum)
        return maxSub

File: Leet Code/53. Maximum Subarray/53. Maximum Subarray.java
class Solution {
     static int maxSubArray(int[] nums) {
        int max_sum = Integer.MIN_VALUE;
        int current_sum = 0;

        for(int i = 0; i<nums.length; i++) {
          
            current_sum = Math.max(current_sum + nums[i], nums[i]);;
            max_sum =  Math.max(max_sum, current_sum);
            
        }
     return max_sum;

    }
}
